Sources/WebSocket/WebSocket.swift

```diff
@@ -6,13 +8,49 @@
 ///      }
 ///
 public final class WebSocket: BasicWorker {
+    
+    /// Available WebSocket modes. Either `Client` or `Server`.
+    public enum Mode {
+        
+        /// Uses socket in `Client` mode
+        case client
+        
+        /// Uses socket in `Server` mode
+        case server
+        
+        /// RFC 6455 Section 5.1
+        /// To avoid confusing network intermediaries (such as intercepting proxies) and
+        /// for security reasons that are further, a client MUST mask all frames that it
+        /// sends to the server.
+        /// The server MUST close the connection upon receiving a frame that is not masked.
+        /// A server MUST NOT mask any frames that it sends to the client.
+        /// A client MUST close a connection if it detects a masked frame.
+        ///
+        /// RFC 6455 Section 5.3
+        /// The masking key is a 32-bit value chosen at random by the client.
+        /// When preparing a masked frame, the client MUST pick a fresh masking
+        /// key from the set of allowed 32-bit values.
+        public func makeMaskKey() -> WebSocketMaskingKey? {
+            switch self {
+            case .client:
+                let buffer = try? CryptoRandom().generateData(count: 4).map { $0 }
+                return buffer.flatMap(WebSocketMaskingKey.init)
+            case .server:
+                return  nil
+            }
+        }
+    }
+    
     /// See `BasicWorker`.
     public var eventLoop: EventLoop {
         return channel.eventLoop
     }
 
     /// Outbound `WebSocketEventHandler`.
     private let channel: Channel
+    
+    /// `WebSocket` processing mode.
+    public private(set) var mode: Mode
```

Comment:
Can a websocket ever change from client to server or vice versa? Seems like this can be `let`.

Reply:
Could this be private?